### PR TITLE
Add support for Events in k8s module

### PR DESCRIPTION
**Pull request by kaisoz**
## Description

This PR adds support for k8s `Events` as stated in #1296 

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [X] Update the docs.
- [X] Run the relevant tests successfully, including pre-commit checks.
- [X] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [ ] Include release notes. If this PR is backwards incompatible, include a migration guide.

## Release Notes (draft)

- [X] Added Event functions.

### Migration Guide

There aren`t backwards incompatible changes

### CODE REVIEW

**Comment by kaisoz** [comment]
This PR currently only adds support for listing events which, in my opinion, is the most used `event`-related operation. However, please let me know if there are other operations you'd like me to implement 😊

**Comment by kaisoz** [comment]
Hi @denis256!

You helped me with other PRs, do you think there's something missing in this one to get it merged? 

Thanks!

**Comment by kaisoz** [comment]
Hi @zachreborn @denis256 ! Do you think this proposal is worth to be merged? 

Thanks!

File: modules/k8s/event.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/gruntwork-io/terratest/modules/testing"
+)
+
+// ListEvents will retrieve the Events in the given namespace that match the given filters and return them. This will fail the
+// test if there is an error.
+func ListEvents(t testing.TestingT, options *KubectlOptions, filters metav1.ListOptions) []corev1.Event {
```
Comment (denis256) [inline modules/k8s/event.go:16]: Will be helpful to have tests for `ListEvents` to track that in the future function will continue to work

Comment (kaisoz) [reply modules/k8s/event.go:16]: Thank you for your review @denis256 . I've done my tests with kubectl and it turns out that the events API doesn't return an error even if you provide the wrong namespace/filters. 

That's why I rewrote the current tests to use the `ListEvents` function and also added a new test which makes sure that `ListEventsE` returns a `nil` error.  What do you think?

**Review by denis256 — APPROVED** [review]

**Comment by denis256** [comment]
Failed tests aren't related with implemented changes
